src/librustc/hir/lowering.rs

```diff
@@ -271,17 +296,23 @@ impl<'a> LoweringContext<'a> {
         o_id.map(|sp_ident| respan(sp_ident.span, sp_ident.node.name))
     }
 
-    fn lower_label(&mut self, id: NodeId, label: Option<Spanned<Ident>>) -> Option<hir::Label> {
-        label.map(|sp_ident| {
-            hir::Label {
-                span: sp_ident.span,
-                name: sp_ident.node.name,
+    fn lower_label(&mut self, label: Option<(NodeId, Spanned<Ident>)>) -> hir::Label {
+        match label {
+            Some((id, label_ident)) => hir::Label {
+                ident: Some(label_ident),
                 loop_id: match self.expect_full_def(id) {
```

Comment:
My concern with that is that there are a lot of places that handle HIR for which it would just be a bug for it to be `None`. `None`/`DUMMY_NODE_ID` should be caught by the loops pass. I'm really starting to think that the loops pass should just be pulled into lowering, if possible.